pos_product_multi_barcode/static/src/js/db.js

```diff
@@ -16,9 +16,6 @@ odoo.define("pos_product_multi_barcode.db", function (require) {
             var res = this._super(products);
             var self = this;
 
-            if (!products instanceof Array) {
-                products = [products];
-            }
```

Comment:
sorry not a Javascript expert here, but are you sure this can be simply removed like that?

Reply:
super is called before, and the original function does the job. So it is not possible to not have an array at this step.

https://github.com/odoo/odoo/blob/16.0/addons/point_of_sale/static/src/js/db.js#L192C1-L194C1

Reply:
Mmm, I'd say it isn't mutable

Reply:
Ow Thanks ! I didn't know. So what is the good way here ? Add an eslint exceptiont ?

Reply:
Well, I'm wrong and is indeed mutable as is part of the `arguments` object: https://eslint.org/docs/latest/rules/no-param-reassign

Reply:
So it is OK like that. Right ?

Reply:
I'd say so

Reply:
> So it is OK like that. Right ?

yes sorry I didn't see the job was done already in super.

Reply:
> Strict mode code doesn’t sync indices of the arguments object with each parameter binding. Therefore, this rule is not necessary to protect against arguments object mutation in ESM modules or other strict mode functions.

I wouldn't remove it. According to that it's only mutable if running old code that hasn't been flagged with `"use strict";`

As we're slowly migrating to ESM, we can't rely on the argument being mutable (which is a bad thing, solved in "modern js")

Reply:
Thanks @ivantodorovich I didn't get that nuance :) Wouldn't the lint rule be  a bit misguiding being that virtually all the code is run in strict mode?

Reply:
Indeed, since we already have the `strict` rule, it seems we could remove the `no-param-reassign` one